src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -998,13 +998,14 @@ namespace winrt::Microsoft::Terminal::Control::implementation
     // Method Description:
     // - Given a copy-able selection, get the selected text from the buffer and send it to the
     //     Windows Clipboard (CascadiaWin32:main.cpp).
-    // - CopyOnSelect does NOT clear the selection
     // Arguments:
     // - singleLine: collapse all of the text to one line
     // - formats: which formats to copy (defined by action's CopyFormatting arg). nullptr
     //             if we should defer which formats are copied to the global setting
+    // - clearSelection: if true, clear the selection. Used for CopyOnSelect.
     bool ControlCore::CopySelectionToClipboard(bool singleLine,
-                                               const Windows::Foundation::IReference<CopyFormat>& formats)
+                                               const Windows::Foundation::IReference<CopyFormat>& formats,
+                                               bool clearSelection)

```

Comment:
This is the only reason why, really: https://github.com/microsoft/terminal/blob/c78818a587badcea162b932c6e17618388f4e80e/src/cascadia/TerminalControl/ControlInteractivity.cpp#L399-L402

Since we're pivoting to clearing the selection after a copy _sometimes_, ControlCore doesn't have knowledge of when that _sometimes_ is, only ControlInteractivity. :/

I can't find the comment, but I think Leonard was suggesting that I make `CopySelectionToClipboard` _never_ clear the selection. Instead, `ControlInteractivity` needs to directly tell `ControlCore` to clear the selection when it needs it. We decided to do that as a follow-up.

Reply:
eh, just did it now haha